lib/geocoder/lookups/olleh.rb

```diff
@@ -157,24 +163,38 @@ def results(query)
 
     def base_url(query)
       case Olleh.check_query_type(query)
-      when "route_search"
-        "https://openapi.kt.com/maps/etc/RouteSearch?params="
-      when "reverse_geocoding"
-        "https://openapi.kt.com/maps/geocode/GetAddrByGeocode?params="
-      when "convert_coord"
-        "https://openapi.kt.com/maps/etc/ConvertCoord?params="
-      when "addr_step_search"
-        "https://openapi.kt.com/maps/search/AddrStepSearch?params="
-      when "addr_nearest_position_search"
-        "https://openapi.kt.com/maps/search/AddrNearestPosSearch?params="
+      when 'addr_local_search'
+        'https://openapi.kt.com/maps/search/km2_LocalSearch?params='
+      when 'route_search'
+        'https://openapi.kt.com/maps/etc/RouteSearch?params='
+      when 'reverse_geocoding'
+        'https://openapi.kt.com/maps/geocode/GetAddrByGeocode?params='
+      when 'convert_coord'
+        'https://openapi.kt.com/maps/etc/ConvertCoord?params='
+      when 'addr_step_search'
+        'https://openapi.kt.com/maps/search/AddrStepSearch?params='
+      when 'addr_nearest_position_search'
+        'https://openapi.kt.com/maps/search/AddrNearestPosSearch?params='
       else #geocoding
-        "https://openapi.kt.com/maps/geocode/GetGeocodeByAddr?params="
+        'https://openapi.kt.com/maps/geocode/GetGeocodeByAddr?params='
       end
     end
 
     def query_url_params(query)
       case Olleh.check_query_type(query)
-      when "route_search"
+      when 'addr_local_search'
+        # option 2 is for sorting results. we are using default.
+        # places is for number of results
+        # sr is for
+        # isaddr for searching address only. excluding building name, etc.
+        hash = {
+          query: URI.encode(query.text),
+          option: "1",
+          places: query.options[:places],
+          sr: query.options[:sr],
+          isaddr: "1"
+        }
+      when 'route_search'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.